components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -508,4 +515,66 @@ protected boolean isPostAuthenticationExtensionCompleted(AuthenticationContext c
         }
     }
 
+    /**
+     * Populate any error information sent from Authenticators to be sent in the Response from the authentication
+     * framework. By default we retrieve the error information from the AuthenticationContext and populate the error
+     * it within the AuthenticationResult as properties.
+     *
+     * @param request
+     * @param response
+     * @param context
+     */
+    protected void populateErrorInformation(HttpServletRequest request,
+                                            HttpServletResponse response,
+                                            AuthenticationContext context) {
+
+        // get the authentication result
+        AuthenticationResult authenticationResult = getAuthenticationResult(request, response, context);
+
+        String errorCode = String.valueOf(context.getProperty(FrameworkConstants.AUTH_ERROR_CODE));
+        String errorMessage = String.valueOf(context.getProperty(FrameworkConstants.AUTH_ERROR_MSG));
+
```

Comment:
Shouldn't we allow user to set username of the failure attempt. There could be use cases we will need that and it would be useful. I think better to allow that also. So someone who needs it can use it. In OAuth level we can set username to error_uri property.

Reply:
Do we need to do this by default? 
In an authentication failure scenario, username could be invalid as well. Also, we are providing an extension point so I feel we should handle failed username in an extension by overriding this method.